xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -477,6 +477,22 @@ unsigned int CAESinkPipewire::AddPackets(uint8_t** data, unsigned int frames, un
 
   stream->QueueBuffer(pwBuffer);
```

Comment:
If this needs a longer time already, it should be calculated into the max blocking time. The below GetTime() might internally not have the precise values for a longer time, therefore guessing is needed, sadly.

See how pulseaudio fixed it: https://freedesktop.org/software/pulseaudio/doxygen/streams.html <- search for interpolate.

We could interpolate here by ourselves ... but easy things first (including QueueBuffer time)
Open bug for pipewire: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/536

Reply:
Adding a 50 ms buffer should never need more than 50 ms of time.

Reply:
updated